ui/app/pages/confirm-approve/confirm-approve.js

```diff
@@ -0,0 +1,138 @@
+import React, { useEffect, useState } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { useParams } from 'react-router-dom'
+import ConfirmTransactionBase from '../confirm-transaction-base'
+import ConfirmApproveContent from './confirm-approve-content'
+import { getCustomTxParamsData } from './confirm-approve.util'
+import { showModal } from '../../store/actions'
+import {
+  getTokenData,
+} from '../../helpers/utils/transactions.util'
+import {
+  calcTokenAmount,
+  getTokenToAddress,
+  getTokenValue,
+} from '../../helpers/utils/token-util'
+import { useTokenTracker } from '../../hooks/useTokenTracker'
+import { getTokens } from '../../ducks/metamask/metamask'
+import {
+  transactionFeeSelector,
+  txDataSelector,
+} from '../../selectors/confirm-transaction'
+import { getCurrentCurrency, getDomainMetadata } from '../../selectors/selectors'
+import { currentNetworkTxListSelector } from '../../selectors/transactions'
+
+export default function ConfirmApprove () {
+  const dispatch = useDispatch()
+  const { id: paramsTransactionId } = useParams()
+  const {
+    id: transactionId,
+    txParams: {
+      to: tokenAddress,
+      data,
+    } = {},
+  } = useSelector(txDataSelector)
+
+  const currentCurrency = useSelector(getCurrentCurrency)
+  const currentNetworkTxList = useSelector(currentNetworkTxListSelector)
+  const domainMetadata = useSelector(getDomainMetadata)
+  const tokens = useSelector(getTokens)
+
+  const transaction = (
+    currentNetworkTxList.find(({ id }) => id === (Number(paramsTransactionId) || transactionId)) || {}
+  )
+  const {
+    ethTransactionTotal,
+    fiatTransactionTotal,
+  } = useSelector((state) => transactionFeeSelector(state, transaction))
+
+  const currentToken = (tokens && tokens.find(({ address }) => tokenAddress === address)) || { address: tokenAddress }
+
+  const { tokensWithBalances } = useTokenTracker([currentToken])
+  const tokenTrackerBalance = tokensWithBalances[0]?.balance || ''
+
+  const tokenSymbol = currentToken?.symbol
+  const decimals = Number(currentToken?.decimals)
+  const tokenData = getTokenData(data)
+  const tokenValue = tokenData && getTokenValue(tokenData.params)
+  const toAddress = tokenData && getTokenToAddress(tokenData.params)
+  const tokenAmount = tokenData && calcTokenAmount(tokenValue, decimals).toString(10)
+
+  const [customPermissionAmount, setCustomPermissionAmount] = useState('')
+
+  useEffect(
+    () => {
+      if (customPermissionAmount) {
+        setCustomPermissionAmount(tokenAmount)
+      }
+    },
+    [tokenAmount]
```

Comment:
This would fail the 'exhaustive deps' rule, but I do want this to only run when `tokenAmount` changes. I could workaround that by using a ref, but that seems like a pain, for no benefit 🤔 

Not sure what convention we want to use here. i.e. maybe we should add a comment explaining why we're not exhaustively checking all deps whenever any are missing.

Reply:
A comment works, sure

Reply:
https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects see the yellow note field. If you do this your values will be stale except when `tokenAmount` changes. The "state" will be a stale object in memory that points to the value it was when the effect was first created. It'll be attached to a react fiber that also holds references to its previous render, and it'll be referenced by a future render. React has done work to mitigate this memory leak but its not rock solid.

I highly recommend following exhaustive deps always.

Reply:
https://github.com/facebook/react/issues/16087

Reply:
It seems perfectly fine for the values to be stale if `tokenAmount` doesn't change - that's OK, right? Unless I'm missing something.

The memory leak is concerning though 👀

Reply:
Reading up on their "better solutions to avoid re-running effects too often", none of them seem appreciably better in this situation. Except maybe that it avoids this memory leak, which I don't quite understand yet.

Reply:
After talking with @brad-decker about this more and hearing him explain the last time he encountered the linked memory leak issue, I now agree that it would be better to err on the side of caution and just do what the `exhaustive-deps` rule says.